persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/RequestToHeadersMapper.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cassandra.impl;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class RequestToHeadersMapper {
+  public static final String TENANT_ID_HEADER_NAME = "tenant_id";
```

Comment:
The `tenant_id` doesn't apply in this context. Instead these should be something like `host` or `publicAddress`.

Reply:
Same for 4.0 and DSE.

Reply:
the `tenant_id` is an actual piece of information that we want to propagate. The caller doesn't care nor should care how it was provided. For now, it is extracted from the `publicAddress` but it may change in the future. We should treat the `tenant_id` header name as the API contract exposed by our code. If we would call it `publicAddress` we would tight couple those concepts. Because it is `tenant_id` we are abstracting away how it is provided. Also, the c2 logic is based on the `tenant_id` header name. wdyt?

Reply:
But if we can deal with just the host that would be much better for users. No one else will use nor understand `tenant_id`

Reply:
`host` would also be misleading in this context, because it would not be a real host name

Reply:
I do not insist, but I do believe that if we want to have cleaner separation of concerns it might be preferable to pass extra auth data via the `AuthenticatedSubject`. We could add an opaque set of properties for simplicity at this point.

Reply:
Feel free to implement that OR change the header name. For both cases, the C2 code needs to be adapted.